mpitests/test_apply.py

```diff
@@ -0,0 +1,594 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`pennylane_lightning_gpu.LightningGPU` device.
+"""
+# pylint: disable=protected-access,cell-var-from-loop
+import math
+from mpi4py import MPI
+import numpy as np
+import pennylane as qml
+import pytest
+from pennylane import DeviceError
+
+try:
+    from pennylane_lightning_gpu.lightning_gpu import CPP_BINARY_AVAILABLE
+    from pennylane_lightning_gpu.lightning_gpu_qubit_ops import MPIManager, DevPool
+    from pennylane_lightning_gpu import LightningGPU
+    import pennylane_lightning_gpu as plg
+
+    if not CPP_BINARY_AVAILABLE:
+        raise ImportError("PennyLane-Lightning-GPU is unsupported on this platform")
+except (ImportError, ModuleNotFoundError):
+    pytest.skip(
+        "PennyLane-Lightning-GPU is unsupported on this platform. Skipping.",
+        allow_module_level=True,
+    )
```

Comment:
I think it is safer to say only that the binary was not found.
We can have a situation where it is supported, but was not compiled, right?

Reply:
Correct!